src/Tasks/Microsoft.NET.Build.Tasks/GenerateRuntimeConfigurationFiles.cs

```diff
@@ -179,7 +180,8 @@ protected override void ExecuteCore()
                                    LockFileTargetLibrary lockFilePlatformLibrary,
                                    bool isFrameworkDependent)
         {
-            runtimeOptions.Tfm = TargetFramework;
+            var targetPlatformSubstring = $"-{NuGetFramework.Parse(TargetFramework).Platform}";
+            runtimeOptions.Tfm = TargetFramework.Split(new[] { targetPlatformSubstring }, StringSplitOptions.RemoveEmptyEntries)[0];
```

Comment:
We shouldn't be parsing the TargetFramework here.  We should try to write our code so that we're never parsing the target framework except in TargetFrameworkInference.targets.  See https://github.com/NuGet/Home/issues/5154

Instead, this code should use the TargetFrameworkMoniker.  Probably we can update the task so that it no longer takes a `TargetFramework` argument.

I also noticed that the following around line 128:

```c#
                ProjectContext projectContext = lockFile.CreateProjectContext(
                    NuGetUtils.ParseFrameworkName(TargetFrameworkMoniker),
                    RuntimeIdentifier,
                    PlatformLibraryName,
                    RuntimeFrameworks,
                    IsSelfContained);
```

It's not directly related to this issue, but I don't think this is going to work with projects with a TargetFramework such as net5.0-windows, since it won't be passing the "-windows" portion in to locate the section of the assets file.  This may be happening in other places in our code too.

Reply:
Oh okay, I guess I misunderstood the issue. So in all cases the tfm in the runtimeconfig should be the TargetFrameworkMoniker, that won't break anything else? 

For line 128, would we want to parse the TargetFramework there instead then?

Reply:
Ping @dsplaisted

Reply:
The `TargetFrameworkMoniker` is in a different format than what we want to write to the runtimeconfig, but yes, we should be able to in all cases parse the `TargetFrameworkMoniker` and then get the NuGet short name version of it to write to the runtimeconfig.

Reply:
I see, I've updated to parse the short name from the TargetFrameworkMoniker.